### PR TITLE
use configuration in the database

### PR DESCRIPTION
**Pull request by cpanato**
Use configuration in the database
Moving the config.json to use the database. Any config you want to set when the pod is starting you can use the extraEnvVars for that

### CODE REVIEW

**Comment by jwilander** [comment]
@cpanato how does this affect exiting helm chart users? Are there easy step someone can follow if they want to move to this new helm chart version?

**Comment by jasonblais** [comment]
@cpanato heads up on above question from Joram

**Comment by cpanato** [comment]
I need to add some docs explaining how to upgrade

**Comment by cpanato** [comment]
Added a readme to upgrade to this new update and did an upgrade test with some custom settings all working.

File: charts/mattermost-enterprise-edition/README.md [hunk 1]
```diff
@@ -205,7 +205,35 @@ To list options for mattermost-push-proxy:
 $ helm inspect values mattermost/mattermost-push-proxy
 ```
 
-# 4. Developing
+# 4. Upgrading to Mattermost Chart Version 1.0.0
+
+This Chart version uses the configuration in the database and environment variables for non changing configuration.
```
Comment (jwilander) [inline charts/mattermost-enterprise-edition/README.md]:
```suggestion
This chart version uses the configuration in the database and environment variables for non-changing configuration.
```

File: charts/mattermost-enterprise-edition/README.md [hunk 2]
```diff
@@ -205,7 +205,35 @@ To list options for mattermost-push-proxy:
...
+
+To upgrade from previous Helm Chart version to 1.0.0 you need to do some steps before applying the new chart
```
Comment (jwilander) [inline charts/mattermost-enterprise-edition/README.md]:
```suggestion
To upgrade from a previous chart version to 1.0.0 you need to follow theses steps before applying the new chart:
```

File: charts/mattermost-enterprise-edition/README.md (hunk 1 above)
Comment (cpanato) [reply charts/mattermost-enterprise-edition/README.md]: done

File: charts/mattermost-enterprise-edition/README.md (hunk 2 above)
Comment (cpanato) [reply charts/mattermost-enterprise-edition/README.md]: done

**Pull request merged**